Classes/model/CoreData/StopLocation.m

```diff
@@ -211,11 +221,15 @@ - (void)clearVisits
 - (NSArray *)alertsIncludingChildren
 {
   if (!_alertsIncludingChildren) {
-    NSMutableArray *alerts = [NSMutableArray array];
-    [alerts addObjectsFromArray:[Alert fetchAlertsForStopLocation:self]];
+    NSMutableArray *alerts = [NSMutableArray array];    
+    if (self.alertHashCodes.count != 0) {
```

Comment:
Doesn't this check make more sense in `fetchAlertsForStopLocation`? Just return an empty array if there are no hash codes.